src/wui/load_or_save_game.cc

```diff
@@ -135,39 +122,41 @@ bool LoadOrSaveGame::compare_map_name(uint32_t rowa, uint32_t rowb) const {
 std::unique_ptr<SavegameData> LoadOrSaveGame::entry_selected() {
 	std::unique_ptr<SavegameData> result(new SavegameData());
 	size_t selections = table_->selections().size();
-	if (selections == 1) {
+	set_tooltips_of_buttons(selections);
+
+	const std::vector<SavegameData> savegames = get_selected_savegames();
+	game_details_.display(savegames);
+	if (selections > 0 && !selection_contains_directory()) {
+		delete_->set_enabled(true);
+	} else {
+		delete_->set_enabled(false);
+		delete_->set_tooltip("");
+	}
+	// TODO(GunChleoc): Take care of the OK button too.
+	if (selections > 0) {
+		result.reset(new SavegameData(savegames[0]));
+	}
+	return result;
+}
+void LoadOrSaveGame::set_tooltips_of_buttons(size_t nr_of_selected_items) const {
+	if (nr_of_selected_items == 1) {
```

Comment:
You could make this more efficient with a switch statement:

case 0
case 1
default

Reply:
Sry, I won't (only if you insist of course)

1. I think it is important for the reader to see the "greater than" to grasp the code's intention quickly. The "default" of your proposed switch would not be the actual default (I suppose it is very rarely that multiple items are selected in this context)
2. it is more code lines

Reply:
Fine by me, this was just a suggestion :)